source/gfoidl.Base64/Internal/Base64Encoder/Base64Encoder.Decode.cs

```diff
@@ -45,6 +45,11 @@ public override int GetDecodedLength(ReadOnlySpan<char> encoded)
             if (Unsafe.Subtract(ref end, 1) == EncodingPad) padding++;
 
             return maxLen - padding;
+        } 
+        //---------------------------------------------------------------------
+        public override int GetMaxDecodedLength(int encodedLength)
+        {
+            return GetDecodedLength(encodedLength);
```

Comment:
Maybe can we remove the padding ?

```suggestion
            return GetDecodedLength(encodedLength) - encodedLength % 3 == 0 ? 0 : 3 - encodedLength% 3;
```

Reply:
I think it not possible for base64 without inspecting the encoded string.

Look at the encoding step, so 

| DataLength | Base64Length | Padding |
| --: | --: |  --: |
| 1 | 4 | 2 |
| 2 | 4 | 1 |
| 3 | 4 | 0 |

So three input values (`dataLength`) map to one `encodedLength`. It's a surjective function, so the reverse is not exactely possible for base64.

That's why the Span-overloads exists, which allow inspection and as there are maximum two padding chars, the code looks like https://github.com/gfoidl/Base64/blob/4ea7bfe0b411386ff3ff8977e291f9ce227a3282/source/gfoidl.Base64/Internal/Base64Encoder/Base64Encoder.Decode.cs#L29-L30

For base64Url it is possible, as no padding is added, so with some calculation the number of padding chars can be determined. It is done like
https://github.com/gfoidl/Base64/blob/4ea7bfe0b411386ff3ff8977e291f9ce227a3282/source/gfoidl.Base64/Internal/Base64UrlEncoder/Base64UrlEncoder.Decode.cs#L534-L539
with some bithack for better performance.

Reply:
After some tweaking on my library, all the use cases with `GetDecodedLength(ReadOnlySpan<byte>()` & `GetDecodedLength(ReadOnlySpan<char>()` are sufficient. Is this PR really required?

The only interest of this PR would be to support `isFinalBlock: false`, that's it?

Reply:
You are right.

So let's close this one, and I'll file a issue for that case and add something for that.
OK?

Reply:
Just came in mind that this case is handled with this PR.

So we should do it. If you want to close I will add this.

Reply:
Sorry for the spamming (it's already late here).

My conclusion is:
* take this PR
* fix up the review comments later -- they work, tests pass, it's just nit or my style